src/package-linker.js

```diff
@@ -380,6 +387,10 @@ export default class PackageLinker {
         linkBinConcurrency,
       );
     }
+
+    for (const [, {pkg}] of flatTree) {
+      await this._warnForMissingBundledDependencies(pkg);
```

Comment:
>linkBinDependencies does not sound like a function that would check bundled dependencies.

Not sure if I agree since it already checks them via that `try`-`catch` block. I don't see the value of indirection here.

>If I thought it would make a big performance difference I probably would have combined them, but I can't imagine that many packages actually have bundled deps, so I'm figuring the vast majority of installs won't see any noticeable performance difference.

I'm a bit concerned about this but I think your reasoning is sound. Still, if there was an "obvious", faster way, I'd prefer that. As my comment above, if these were completely separate things, I'd be okay but looks like we just defer the work we could have done earlier without a very clear separation and paying the cost of separation. Or I'm missing something here.